### PR TITLE
Don't dispose generated classes when Zombies is enabled

### PR DESCRIPTION
**Pull request by ncooke3**
This is the original fix that checks the env vars for Zombies being enabled. If Zombies is enabled, then the generated class is not disposed

Fix https://github.com/firebase/firebase-ios-sdk/issues/8321

### CODE REVIEW

**Comment by ncooke3** [comment]
Should [this](https://github.com/google/GoogleUtilities/pull/57/files#diff-0b49698c5a97ff29bc94a0a90b319a45f1b5c021b285e0c8b2a49a9edf73fed5R197) disposal be guarded with the zombies check as well? I haven't been able to ever trigger that check.

File: GoogleUtilities/ISASwizzler/GULObjectSwizzler.m [hunk 1]
```diff
@@ -163,8 +163,16 @@ - (void)dealloc {
...
-      // now.
-      objc_disposeClassPair(_generatedClass);
+      // now (unless the Zombies instrument is enabled).
+
+      // When the Zombies instrument is enabled, a zombie is created for the
+      // swizzled object upon deallocation. Because this zombie subclasses
+      // the generated class, the swizzler should not dispose it.
+      NSDictionary *environment = [[NSProcessInfo processInfo] environment];
+      BOOL zombiesEnabled = [[environment objectForKey:@"NSZombieEnabled"] boolValue];
+      if (!zombiesEnabled) {
```
Comment (maksymmalyhin) [inline GoogleUtilities/ISASwizzler/GULObjectSwizzler.m:173]: I think we should apply this condition to `objc_disposeClassPair(generatedClass);` used below as well.

**Review by maksymmalyhin — APPROVED** [review]
LGTM, one comment.

**Comment by ncooke3** [comment]
[Failed GHA](https://github.com/google/GoogleUtilities/pull/57/checks?check_run_id=3487918814) doesn't look related but Im unsure

**Comment by paulb777** [comment]
Probably a timing flake: " Asynchronous wait failed: Exceeded timeout of 10 seconds, with unfulfilled expectations: "Expect block is called"

**Review by maksymmalyhin — APPROVED** [review]
LGTM on relevant CI green. Thank you!

**Pull request merged**